substrate/frame/examples/kitchensink/src/lib.rs

```diff
@@ -226,6 +226,32 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// A call that is specially authorized.
+		/// Authorized call can be dispatched by anybody without requiring any signature or fee.
+		#[pallet::call_index(1)]
+		#[pallet::authorize(|
+			_source: TransactionSource,
+			new_foo: &u32,
+		| -> TransactionValidityWithRefund {
+			if *new_foo == 42 {
+				let refund = Weight::zero();
+				let validity = ValidTransaction::default();
```

Comment:
Currently this is not correct: A valid transaction needs to provide something, in the `ValidTransaction` struct.

Maybe we can by default provide the call, or otherwise require users to always specify what is provides.
But then I think there is some risk: if a call is valid for multiple parameters (e.g. `fn reset_storage(upper_bound: u32)` would be valid for any `upper_bound` superior to some value N). then we can spam the node with all those calls.

Same as validate unsigned it is better to require user to define the `provides`.